deepchecks/vision/checks/data_integrity/property_label_correlation.py

```diff
@@ -0,0 +1,195 @@
+    """
+
+    def __init__(
+            self,
+            image_properties: Optional[List[Dict[str, Any]]] = None,
+            n_top_properties: int = 3,
+            per_class: bool = True,
+            random_state: int = None,
+            min_pps_to_show: float = 0.05,
+            ppscore_params: dict = None,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.image_properties = image_properties if image_properties else default_image_properties
+
+        self.min_pps_to_show = min_pps_to_show
+        self.per_class = per_class
+        self.n_top_properties = n_top_properties
+        self.random_state = random_state
+        self.ppscore_params = ppscore_params or {}
+        self._properties_results = defaultdict(list)
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind):
+        """Calculate image properties for train or test batches."""
+        data_for_properties, target = calc_properties_for_property_label_correlation(
+            context, batch, dataset_kind, self.image_properties)
+
+        self._properties_results['target'] += target
+
+        for prop_name, property_values in data_for_properties.items():
+            self._properties_results[prop_name].extend(property_values)
+
+    def compute(self, context: Context, dataset_kind: DatasetKind) -> CheckResult:
+        """Calculate the PPS between each property and the label.
+
+        Returns
+        -------
+        CheckResult
+            value: dictionaries of PPS values.
+            display: bar graph of the PPS of each feature.
+        """
+        df_props = pd.DataFrame(self._properties_results)
+
+        # PPS task type is inferred from label dtype. For most computer vision tasks, it's safe to assume that unless
+        # the label is a float, then the task type is not regression and thus the label is cast to object dtype.
+        # For the known task types (object detection, classification), classification is always selected.
+
+        col_dtype = 'object'
+        if context.train.task_type == TaskType.OTHER:
+            if is_float_column(df_props['target']):
+                col_dtype = 'float'
+        elif context.train.task_type not in (TaskType.OBJECT_DETECTION, TaskType.CLASSIFICATION):
+            raise ModelValidationError(
+                f'Check must be explicitly adopted to the new task type {context.train.task_type}, so that the '
+                f'label type used by the PPS predictor would be appropriate.')
+
+        df_props['target'] = df_props['target'].astype(col_dtype)
+
+        df_pps = pps.predictors(df=df_props, y='target', random_seed=self.random_state,
+                                **self.ppscore_params)
+        s_ppscore = df_pps.set_index('x', drop=True)['ppscore']
+
+        if context.with_display:
+            top_to_show = s_ppscore.head(self.n_top_properties)
+
+            fig = get_pps_figure(per_class=False, n_of_features=len(top_to_show))
```

Comment:
Seems like self.per_class is not used. I see that we didn't implement a per_class option for this check in tabular.

What I suggest in that case is removing the per class option for now from this check, and add a future issue about adding it fully.

Reply:
https://github.com/deepchecks/deepchecks/issues/1903